rick-and-morty/src/CharacterList.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import Character from './Character';
+import CharacterPageNavigator from './CharacterPageNavigator';
+
+export default class CharacterList extends React.Component {
+  state = {
+    isLoading: false,
+    response: {
+      info: {},
+      results: []
+    },
+    characterListPageNumber: 1,
+    pages: 1
+  }
+
+  getCharacters = () => {
+    this.setState({ isLoading: true }, () => {
+      fetch(`https://rickandmortyapi.com/api/character/?page=${this.state.characterListPageNumber}`)
+        .then(result => result.json())
+        .then(data => this.setState({ response: data, isLoading: false }))
+    });
+  }
+
+  componentDidMount = () => {
+    this.getCharacters();
+  }
+
+  setCharacterListPageNumber = (characterListPageNumber) => {
+    this.setState({
+      characterListPageNumber: Number(characterListPageNumber)
+    }, this.getCharacters())
```

Comment:
Should be `this.getCharacters` or `() => this.getCharacters()` here, otherwise `getCharacters` will be executed immediately.